src/org/labkey/remoteapi/assay/Protocol.java

```diff
@@ -0,0 +1,415 @@
+        JSONArray domains = new JSONArray();
+        result.put("domains", domains);
+        for (Domain domain : _domains)
+            domains.add(domain.toJSONObject(true));
+
+        if (_allowBackgroundUpload != null)
+            result.put("allowBackgroundUpload", _allowBackgroundUpload);
+        if (_backgroundUpload != null)
+            result.put("backgroundUpload", _backgroundUpload);
+        if (_allowEditableResults != null)
+            result.put("allowEditableResults", _allowEditableResults);
+        if (_editableResults != null)
+            result.put("editableResults", _editableResults);
+        if (_allowQCStates != null)
+            result.put("allowQCStates", _allowQCStates);
+        if (_qcEnabled != null)
+            result.put("qcEnabled", _qcEnabled);
+        if (_allowSpacesInPath != null)
+            result.put("allowSpacesInPath", _allowSpacesInPath);
+        if (_allowTransformationScript != null)
+            result.put("allowTransformationScript", _allowTransformationScript);
+        if (_saveScriptFiles != null)
+            result.put("saveScriptFiles", _saveScriptFiles);
+        if (_editableRuns != null)
+            result.put("editableRuns", _editableRuns);
+        if (_autoCopyTargetContainerId != null)
+            result.put("autoCopyTargetContainerId", _autoCopyTargetContainerId);
+
+        if (_availableDetectionMethods != null)
+        {
+            JSONArray detectionMethods = new JSONArray();
+            result.put("availableDetectionMethods", detectionMethods);
+            for (String detectionMethod : _availableDetectionMethods)
+                detectionMethods.add(detectionMethod);
+        }
+        if (_selectedDetectionMethod != null)
+            result.put("selectedDetectionMethod", _selectedDetectionMethod);
+
+        if (_availableMetadataInputFormats != null)
+            result.put("availableMetadataInputFormats", _availableMetadataInputFormats);
+        if (_selectedMetadataInputFormat != null)
+            result.put("selectedMetadataInputFormat", _selectedMetadataInputFormat);
+
+        if (_availablePlateTemplates != null)
+        {
+            JSONArray plateTemplates = new JSONArray();
+            result.put("availablePlateTemplates", plateTemplates);
+            for (String plateTemplate : _availablePlateTemplates)
+                plateTemplates.add(plateTemplate);
+        }
+        if (_selectedPlateTemplate != null)
+            result.put("seletedPlateTemplate", _selectedPlateTemplate);
+
+        if (_protocolParameters != null)
+            result.put("protocolParameters", _protocolParameters);
+
+        if (_protocolTransformScripts != null)
+        {
+            JSONArray transformScripts = new JSONArray();
+            result.put("protocolTransformScripts", transformScripts);
+            for (String script : _protocolTransformScripts)
+                transformScripts.add(script);
+        }
```

Comment:
This one fits with the comment above related to if we need to include the static parameters in the JSON object here (so I think the set would be the ones prefixed with "allow", prefixed with "available", and protocolTransformScripts. I don't know what protocolParameters are, might be worth checking with @labkey-klum